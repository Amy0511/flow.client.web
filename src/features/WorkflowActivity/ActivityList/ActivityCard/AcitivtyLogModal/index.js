import React, { Component } from "react";
import PropTypes from "prop-types";
import axios from "axios";
import Modal from "@boomerang/boomerang-components/lib/Modal";
import ModalFlow from "@boomerang/boomerang-components/lib/ModalFlow";
import ModalContentBody from "@boomerang/boomerang-components/lib/ModalContentBody";
import { ACTIVITY_STATUSES } from "Constants/activityStatuses";
import { BASE_SERVICE_URL, REQUEST_STATUSES } from "Config/servicesConfig";

export default class ActivityLogModal extends Component {
  static propTypes = {
    activityId: PropTypes.string.isRequired,
    activityStatus: PropTypes.string.isRequired,
    taskId: PropTyes.string.isRequired
  };

  state = {
    log: ""
  };
  componentDidMount() {
    if (this.props.activityStatus !== ACTIVITY_STATUSES.COMPLETED) {
      this.interval = setInterval(this.fetchLog, 3000);
    }
  }

  fetchLog = () => {
    axios
      .get(`${BASE_SERVICE_URL}/${activityId}/log/${taskId}`)
      .then(response => this.setState({ log: response.data.message }));
  };

  render() {
    return (
      <Modal
        ModalTrigger={() => <div>View Log</div>}
        modalContent={(closeModal, ...rest) => (
          <ModalFlow closeModal={closeModal} theme="bmrg-white" headerTitle="Activity Log">
            <ModalContentBody>Content</ModalContentBody>
          </ModalFlow>
        )}
      />
    );
  }
}
