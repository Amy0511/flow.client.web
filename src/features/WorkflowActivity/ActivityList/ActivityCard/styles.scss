@import "~@boomerang/boomerang-styles/lib/colors";

.c-activities-card {
  background-color: $white;
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  height: 7.5rem;
  border: solid 0.05rem $teal-20;
  border-radius: 0.25rem;
  margin: 1.5rem 1rem;
  cursor: pointer;
  &.--completed {
    border-color: $teal-20;
  }
  &.--failure {
    border-color: $red-20;
  }
  &.--inProgress {
    border-color: $cerulean-20;
  }
}
.c-activities-row__item-workflow {
  display: flex;
  align-items: center;
}
.c-activities-row__item-activity {
  display: flex;
  align-items: center;
  justify-content: flex-end;
}

.c-activity-card {
  background-color: $white;
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 50rem;
  height: 7.5rem;
  border: solid 0.125rem $teal-20;
  border-radius: 0.5rem;
  margin: 1.5rem 1rem;
  &.--success {
    border-color: $teal-20;
  }
  &.--failure {
    border-color: $red-20;
  }
  &.--inprogess {
    border-color: $cerulean-20;
  }
}
.c-activity-card__workflow {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
}
.c-activity-card__workflow--left {
  display: flex;
}
.c-activity-card__workflow--right {
  display: flex;
  margin-top: 0.75rem;
  margin-right: 2rem;
}
.c-activity-card__workflow-first-column {
  width: 17rem;
  margin-right: 4rem;
}
.c-activity-card__trigger-info {
  display: flex;
}
.c-activity-card__workflow-info {
  margin-left: 0.5rem;
  justify-content: center;
  display: flex;
  align-items: flex-start;
  flex-direction: column;
  min-width: 12rem;
}

.c-activity-card__workflow-column {
  margin-top: 1rem;
  display: flex;
}
.c-activity-card__workflow-trigger-label {
  display: flex;
  flex-direction: column;
}
.c-activity-card__workflow-trigger-info {
  display: flex;
  flex-direction: column;
  margin-left: 1rem;
}

.c-activity-card__icon {
  margin-left: 1.5rem;
  width: 5rem;
  display: flex;
  align-items: center;
  justify-content: flex-start;
}
.b-activity-card__icon {
  width: 5rem;
  height: 4rem;
}
.b-activity-card__description {
  color: $gray-60;
  font-size: 1.1rem;
}
.b-activity-card__name {
  font-size: 1.25rem;
  font-weight: 600;
  margin-bottom: 0.5rem;
}
.b-activity-card__team-name {
  font-size: 1rem;
  font-weight: 600;
  margin-bottom: 0.75rem;
  display: block;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  width: 13rem;
  min-height: 1rem;
}
.b-activity-card__trigger-label {
  font-size: 1rem;
  font-weight: 600;
  margin-bottom: 0.75rem;
  width: 6rem;
}
.b-activity-card__label {
  width: 6rem;
  font-size: 1rem;
  font-weight: 600;
}
.b-activity-card__team-name,
.b-activity-card__trigger {
  margin-bottom: 0.75rem;
  font-weight: initial;
  color: $gray-60;
}
.b-activity-card__autor {
  margin-bottom: 0.5rem;
  color: $gray-60;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  width: 10rem;
}
.b-activity-card__status-icon {
  margin-right: 0.5rem;
}
.b-activity-card__data {
  margin-left: 2rem;
  display: flex;
  align-items: center;
  color: $gray-60;
}
.c-activity-card__activity {
  margin-top: 1rem;
  margin-right: 3rem;
}
.c-activity-card__label {
  display: flex;
  justify-content: flex-start;
  margin-bottom: 0.75rem;
}
