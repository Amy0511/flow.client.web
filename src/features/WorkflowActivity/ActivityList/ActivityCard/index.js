import React, { Component } from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import moment from "moment";
import getHumanizedDuration from "@boomerang/boomerang-utilities/lib/getHumanizedDuration";
import { ACTIVITY_STATUSES_TO_TEXT, ACTIVITY_STATUSES_TO_ICON } from "Constants/activityStatuses";
import imgs from "Assets/icons";
import "./styles.scss";

class ActivityCard extends Component {
  static propTypes = {
    activity: PropTypes.object.isRequired,
    match: PropTypes.object.isRequired
  };

  render() {
    const {
      description,
      creationDate,
      icon,
      duration = 0,
      id,
      initiatedByUserName,
      status,
      teamName,
      trigger,
      workflowId,
      workflowName
    } = this.props.activity;
    return (
      <Link
        className={`c-activities-card --${status}`}
        to={{
          pathname: `/activity/${workflowId}/execution/${id}`,
          state: { fromUrl: this.props.match.url, fromText: "Activity" }
        }}
      >
        <div className="c-activity-card-workflow">
          <div className="c-activity-card-section--left">
            <div className="c-activity-card__icon">
              <img className="b-activity-card__icon" src={imgs[icon ? icon : "docs"]} alt="Worklflow icon" />
            </div>
            <div className="c-activity-card-workflow-info">
              <label className="b-activity-card__name">{workflowName}</label>
              <label className="b-activity-card__description">{description}</label>
            </div>
          </div>
          <div className="c-activity-card-section--right">
            <ul className="c-activity-card-execution-info --first">
              <li className="b-activity-card__row">
                <label className="b-activity-card__label">Team</label>
                <label className="b-activity-card__data">{teamName}</label>
              </li>
              <li className="b-activity-card__row">
                <label className="b-activity-card__label">Trigger</label>
                <label className="b-activity-card__data">
                  {trigger ? trigger.slice(0, 1).toUpperCase() + trigger.slice(1) : "---"}
                </label>
              </li>
              <li className="b-activity-card__row">
                <label className="b-activity-card__label">Initiated by</label>
                <label className="b-activity-card__data">{initiatedByUserName || "---"}</label>
              </li>
            </ul>
            <ul className="c-activity-card-execution-info">
              <li className="b-activity-card__row">
                <label className="b-activity-card__label">Start Time</label>
                <label className="b-activity-card__data">{moment(creationDate).format("YYYY-MM-DD hh:mm A")}</label>
              </li>
              <li className="b-activity-card__row">
                <label className="b-activity-card__label">Duration</label>
                <label className="b-activity-card__data">
                  {duration ? getHumanizedDuration(parseInt(duration / 1000, 10)) : "---"}
                </label>
              </li>
              <li className="b-activity-card__row">
                <label className="b-activity-card__label"> Status</label>
                <label className="b-activity-card__data">
                  <img
                    className="b-activity-card__status-icon"
                    src={ACTIVITY_STATUSES_TO_ICON[status ? status : "notstarted"]}
                    alt={`Status ${status}`}
                  />
                  {ACTIVITY_STATUSES_TO_TEXT[status ? status : "notstarted"]}
                </label>
              </li>
            </ul>
          </div>
        </div>
      </Link>
    );
  }
}

export default ActivityCard;
