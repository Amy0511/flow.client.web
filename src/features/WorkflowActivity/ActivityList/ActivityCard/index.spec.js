import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router";
import ActivityCard from "./index";

const history = {};
// const activity = {
//   "last": false,
//     "totalPages": 9,
//     "totalElements": 90,
//     "size": 10,
//     "number": 0,
//     "numberOfElements": 10,
//     "first": true,
//     "sort": [
//       {
//         "direction": "DESC",
//         "property": "creationDate",
//         "ignoreCase": false,
//         "nullHandling": "NATIVE",
//         "descending": true,
//         "ascending": false
//       }
//     ],
//     "records":[
//       {
//         "creationDate": 1541070397344,
//         "duration": 13886,
//         "id": "5bdade3d37dc4700011b06c3",
//         "initiatedBy": "5bc7b126f7856000012cd95d",
//         "status": "completed",
//         "workflowId": "5bd9d8ab7eb44d0001772e64",
//         "workflowRevisionid": "5bd9e1cfa40f5d0001249bfa",
//         "description": "file ingestion",
//         "teamName": "Isa's Team",
//         "icon": "flow",
//         "shortDescription": "file ingestion",
//         "workflowName": "Business Process Test"
//       }
//     ]
// };
const activity = {
  creationDate: 1541070397344,
  duration: 13886,
  id: "5bdade3d37dc4700011b06c3",
  initiatedBy: "5bc7b126f7856000012cd95d",
  status: "completed",
  workflowId: "5bd9d8ab7eb44d0001772e64",
  workflowRevisionid: "5bd9e1cfa40f5d0001249bfa",
  description: "file ingestion",
  teamName: "Isa's Team",
  icon: "flow",
  shortDescription: "file ingestion",
  workflowName: "Business Process Test"
};

describe("ActivityCard --- Snapshot", () => {
  it("Capturing Snapshot of ActivityCard", () => {
    const renderedValue = renderer.create(
      <MemoryRouter>
        <ActivityCard activity={activity} history={history} />
      </MemoryRouter>
    );
    expect(renderedValue).toMatchSnapshot();
  });
});

describe("ActivityCard --- Shallow render", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <MemoryRouter>
        <ActivityCard activity={activity} history={history} />
      </MemoryRouter>
    );
  });

  it("Render the DUMB component", () => {
    expect(wrapper.length).toEqual(1);
  });
});
