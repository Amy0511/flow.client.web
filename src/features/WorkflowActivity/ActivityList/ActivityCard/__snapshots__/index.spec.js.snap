// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityCard --- Snapshot Capturing Snapshot of ActivityCard 1`] = `
<a
  className="c-activities-card --completed"
  href="/activity/5bd9d8ab7eb44d0001772e64/execution/5bdade3d37dc4700011b06c3"
  onClick={[Function]}
>
  <div
    className="c-activity-card__workflow"
  >
    <div
      className="c-activity-card__icon"
    >
      <img
        alt="icon"
        className="b-activity-card__icon"
        src="flow.svg"
      />
    </div>
    <div
      className="c-activity-card__workflow-info"
    >
      <label
        className="b-activity-card__team-name"
      >
        Isa's Team
      </label>
      <label
        className="b-activity-card__name"
      >
        Business Process Test
      </label>
      <label
        className="b-activity-card__description"
      >
        file ingestion
      </label>
    </div>
  </div>
  <div
    className="c-activity-card__activity"
  >
    <div
      className="c-activity-card__label"
    >
      <label
        className="b-activity-card__label"
      >
        Start Time
      </label>
      <label
        className="b-activity-card__data"
      >
        2018-11-01 08:06 AM
      </label>
    </div>
    <div
      className="c-activity-card__label"
    >
      <label
        className="b-activity-card__label"
      >
        Duration
      </label>
      <label
        className="b-activity-card__data"
      >
        13 seconds
      </label>
    </div>
    <div
      className="c-activity-card__label"
    >
      <label
        className="b-activity-card__label"
      >
         Status
      </label>
      <label
        className="b-activity-card__data"
      >
        <img
          alt="Status completed"
          className="b-activity-card__status-icon"
          src="passed.svg"
        />
        Succeeded
      </label>
    </div>
  </div>
</a>
`;
